### ISSUE TITLE
Update Storybook support for storybook@latest

### ISSUE THREAD

**Issue by mnquintana** [solidui/solid#430]
The Storybook API has [changed quite a lot](https://storybook.js.org/docs/react/writing-stories/introduction) since [Solid's Storybook docs](https://github.com/solidui/solid/blob/main/documentation/storybook.md) were written in 2019. It would be really helpful (and I think go a loooong way toward helping drive adoption, long-term) if Solid could:

1. Update the docs to reflect the latest Storybook API (hopefully everything still works with 6.2+?)
2. Abstract all these manual setup steps away with an official Solid framework plugin for Storybook (docs for how to do this here: https://storybook.js.org/docs/react/api/new-frameworks)

Storybook's become a pretty essential piece of the UI development pipeline, and at least for me, official support for it is a prerequisite for seriously considering any frontend frameworks. I miiiight be able to help with setting this up for Solid, although I haven't tried adding support for a new framework to Storybook before. 😅

**Comment by rturnq** [comment]
I created [storybook-solid](https://github.com/rturnq/storybook-solid/tree/master/app/solid) a little while ago but haven't had the time to really try it out much. This follows the guide from Storybook you linked. It's not ideal yet as it would need buy-in from Storybook to add it as a community supported framework to they cli tool, but it saves you from having to write boilerplate.

It would be great to get some feedback on this and once it has been tested out a bit more, then maybe we could get it added to the docs.

**Comment by ryansolid** [comment]
Yeah I'm all for this. I just haven't had the reason to look into it. I was hoping that someone would update the docs at some point. I'm not sure if stuff is different now. If anyone wants to help putting such a plugin together I'd support it and do what is needed to get it recognized. I'm just not working in Storybook personally.

**Comment by ryansolid** [comment]
On further review I think we are best setting up a repo for a plugin following the steps and not bothering with docs here. It probably should be managed independently.

**Comment by mnquintana** [comment]
Sounds good to me! Let me know if there's anything I can help with.

**Comment by grenierdev** [comment]
I was able to set up Storybook (v6.3.4) in my project using the _@storybook/html_ mode and without the #553 bug.

Turns out we just need to create our own root and call `render` on this new root. In `.storybook/preview.js`, add these lines and your good to go!

```js
import { render } from "solid-js/web";

export const decorators = [
  Story => {
     const root = document.getElementById("root");
     const solid = document.createElement("div");
	 
     solid.setAttribute('id', 'solid-root');  
     root.appendChild(solid);
     render(Story, solid);
     return solid;
     // return createRoot(() => Story()); // do not work correctly https://github.com/solidjs/solid/issues/553
  }
]
```

**Comment by arjun-g** [comment]
Anyone had success with using mdx with solidjs ?

**Comment by pakholeung37** [comment]
If someone wants to use storybook with vite, this would help
https://codesandbox.io/s/qx2r0?file=/.storybook/main.js:0-404
